pkg/retriever/graphsyncretriever.go

```diff
@@ -229,78 +227,60 @@ func runRetrievalCandidate(
 	var retrievalErr error
 	var done func()
 
-	retrieval.sendEvent(events.Started(req.RetrievalID, phaseStartTime, types.QueryPhase, candidate))
-
-	// run the query phase
-	onConnected := func() {
-		retrieval.sendEvent(events.Connected(req.RetrievalID, phaseStartTime, types.QueryPhase, candidate))
-	}
-	queryResponse, queryErr := queryPhase(ctx, cfg, client, timeout, candidate, onConnected)
-	if queryErr != nil {
-		retrieval.sendEvent(events.Failed(req.RetrievalID, phaseStartTime, types.QueryPhase, candidate, queryErr.Error()))
+	retrieval.sendEvent(events.Started(cfg.Clock.Now(), req.RetrievalID, phaseStartTime, types.RetrievalPhase, candidate))
+	connectCtx := ctx
+	if timeout != 0 {
+		var timeoutFunc func()
+		connectCtx, timeoutFunc = context.WithDeadline(ctx, time.Now().Add(timeout))
+		defer timeoutFunc()
 	}
 
-	// treat QueryResponseError as a failure
-	if queryResponse != nil && queryResponse.Status == retrievaltypes.QueryResponseError {
-		retrieval.sendEvent(events.Failed(req.RetrievalID, phaseStartTime, types.QueryPhase, candidate, queryResponse.Message))
-		queryResponse = nil
-	}
+	err := client.Connect(connectCtx, candidate.MinerPeer)
 
-	if queryResponse != nil {
-		retrieval.sendEvent(events.QueryAsked(req.RetrievalID, phaseStartTime, candidate, *queryResponse))
-		if queryResponse.Status != retrievaltypes.QueryResponseAvailable ||
-			(cfg.IsAcceptableQueryResponse != nil && !cfg.IsAcceptableQueryResponse(candidate.MinerPeer.ID, req, queryResponse)) {
-			queryResponse = nil
+	if err != nil {
+		if ctx.Err() == nil { // not cancelled, maybe timed out though
+			log.Warnf("Failed to connect to miner %s: %v", candidate.MinerPeer.ID, err)
+			retrievalErr = fmt.Errorf("%w: %v", ErrConnectFailed, err)
+			retrieval.sendEvent(events.Failed(cfg.Clock.Now(), req.RetrievalID, phaseStartTime, types.RetrievalPhase, candidate, retrievalErr.Error()))
 		}
-	}
-
-	if queryResponse != nil {
-		retrieval.sendEvent(events.QueryAskedFiltered(req.RetrievalID, phaseStartTime, candidate, *queryResponse))
-
+	} else {
+		retrieval.sendEvent(events.Connected(cfg.Clock.Now(), req.RetrievalID, phaseStartTime, types.RetrievalPhase, candidate))
 		// if query is successful, then wait for priority and execute retrieval
-		done = retrieval.waitQueue.Wait(&queryCandidate{queryResponse, time.Since(phaseStartTime)})
+		done = retrieval.waitQueue.Wait(connectCandidate{
```

Comment:
k, so the new prioritisation mechanism is—first to get past `Connect` proceeds to retrieval attempt; the rest queue up and are ordered based on their indexer metadata followed by the speed/order in which they got into the queue.

👍

Reply:
yes. for now.